src/Chapter4.hs

```diff
@@ -472,10 +486,13 @@ Implement the Applicative instance for our 'Secret' data type from before.
 -}
 instance Applicative (Secret e) where
     pure :: a -> Secret e a
-    pure = error "pure Secret: Not implemented!"
+    pure a = Reward a
 
     (<*>) :: Secret e (a -> b) -> Secret e a -> Secret e b
-    (<*>) = error "(<*>) Secret: Not implemented!"
+    (Reward f) <*> (Reward x) = Reward (f x)
+    (Trap a) <*> _ = Trap a
+    _ <*> (Trap b) = Trap b
```

Comment:
Correct implementation! You can notice that the first and the last line look very similar to the `Functor` instance for `Reward`, so you can reuse it here :slightly_smiling_face: